File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 6/11/2026, 11:38:38 AM
class Solution {
public:
    int maxArea(vector<int>& arr) {
        int n = arr.size();
        int i=0;
        int j=n-1;
        int marea = 0;
        while(i<j){
            int cura = min(arr[j], arr[i])*(j-i);
            marea = max(marea, cura);
            if(arr[i]<arr[j]){
                i++;
            }else{
                j--;
            }
        }

        return marea;
        
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 6/11/2026, 11:38:44 AM
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1

        maxar = 0
        while(l<r):
            maxar = max(maxar, (r-l)*(min(height[l],height[r])))
            if(height[l]<height[r]):
                l += 1
            else:
                r -= 1
        return maxar

        